test/byContract/registries/resources.test.ts

```diff
@@ -0,0 +1,179 @@
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { deployments, ethers, expect } from 'hardhat';
+import { loadHardhatFixtureDeployment } from '../../../scripts/hardhat-fixture-deployer';
+import {
+    App,
+    AppRegistry,
+    AppRegistry__factory,
+    App__factory,
+    Dataset,
+    DatasetRegistry,
+    DatasetRegistry__factory,
+    Dataset__factory,
+    ENSRegistry,
+    ENSRegistry__factory,
+    IexecInterfaceNative,
+    IexecInterfaceNative__factory,
+    PublicResolver__factory,
+    ReverseRegistrar__factory,
+    Workerpool,
+    WorkerpoolRegistry,
+    WorkerpoolRegistry__factory,
+    Workerpool__factory,
+} from '../../../typechain';
+import { getIexecAccounts } from '../../../utils/poco-tools';
+const constants = require('../../../utils/constants');
+
+describe('Ressources', () => {
+    let proxyAddress: string;
+    let [iexecPoco, iexecPocoAsAdmin]: IexecInterfaceNative[] = [];
+    let [iexecAdmin, appProvider, datasetProvider, scheduler, anyone]: SignerWithAddress[] = [];
+
+    let ensRegistry: ENSRegistry;
+    let appRegistry: AppRegistry;
+    let datasetRegistry: DatasetRegistry;
+    let workerpoolRegistry: WorkerpoolRegistry;
+
+    let app: App;
+    let dataset: Dataset;
+    let workerpool: Workerpool;
+
+    beforeEach(async () => {
+        proxyAddress = await loadHardhatFixtureDeployment();
+        await loadFixture(initFixture);
+    });
+
+    async function initFixture() {
+        ({ iexecAdmin, appProvider, datasetProvider, scheduler, anyone } =
+            await getIexecAccounts());
+        const ensRegistryAddress = (await deployments.get('ENSRegistry')).address;
+        ensRegistry = ENSRegistry__factory.connect(ensRegistryAddress, anyone);
+
+        iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, anyone);
+        iexecPocoAsAdmin = iexecPoco.connect(iexecAdmin);
+
+        appRegistry = AppRegistry__factory.connect(await iexecPoco.appregistry(), anyone);
+        datasetRegistry = DatasetRegistry__factory.connect(
+            await iexecPoco.datasetregistry(),
+            anyone,
+        );
+        workerpoolRegistry = WorkerpoolRegistry__factory.connect(
+            await iexecPoco.workerpoolregistry(),
+            anyone,
+        );
+    }
+
+    describe('App', () => {
+        const createAppArgs = [
+            `App`,
+            'DOCKER',
+            constants.MULTIADDR_BYTES,
+            ethers.utils.keccak256(ethers.utils.toUtf8Bytes(`Content of my app`)),
+            '0x1234',
+        ] as [string, string, BytesLike, BytesLike, BytesLike];
+
+        beforeEach(async () => {
+            const appAddress = await appRegistry.callStatic.createApp(
+                appProvider.address,
+                ...createAppArgs,
+            );
+            await appRegistry.createApp(appProvider.address, ...createAppArgs);
+            app = App__factory.connect(appAddress, appProvider);
```

Comment:
Load app with anyone, then explicit .`connect(appProvider)` elsewhere

Reply:
Will do it later